include/tsid/trajectories/trajectory-base.hpp

```diff
@@ -18,43 +18,100 @@
 #ifndef __invdyn_trajectory_base_hpp__
 #define __invdyn_trajectory_base_hpp__
 
+#include <pinocchio/spatial/se3.hpp>
+
+#include "tsid/deprecation.hpp"
+#include "tsid/macros.hpp"
 #include "tsid/math/fwd.hpp"
+#include "tsid/math/utils.hpp"
 
 #include <string>
 
 namespace tsid
 {
   namespace trajectories
   {
-    
+
+    typedef Eigen::Map<const Eigen::Matrix<double, 3, 3>> MapMatrix3;
+
     class TrajectorySample
     {
     public:
       EIGEN_MAKE_ALIGNED_OPERATOR_NEW
-      
-      math::Vector pos, vel, acc;
+
+      // TODO rename pos, vel, acc → value, derivative, second_derivative
+      DEPRECATED math::Vector pos, vel, acc;
+
+TSID_DISABLE_WARNING_PUSH
+TSID_DISABLE_WARNING_DEPRECATED
+      // getters / setters with updated names for math::Vector
+      const math::Vector & getValue() const { return pos; }
+      const math::Vector & getDerivative() const { return vel; }
+      const math::Vector & getSecondDerivative() const { return acc; }
+      void setValue(const math::Vector & value) { pos = value; }
+      void setDerivative(const math::Vector & derivative) { vel = derivative; }
+      void setSecondDerivative(const math::Vector & second_derivative) { acc = second_derivative; }
+
+      // getters / setters with updated names for pinocchio::SE3
+      void getValue(pinocchio::SE3 & value) const
```

Comment:
If I remember correctly the initial design choices, I think this class didn't contain any specific reference to SE3 because it's the base class. Everything specific to SE3 was added in the derived class. Is there a particular reason why it's necessary to add this SE3-specific code here?

Reply:
It's the Sample class, which is a member of the Base class. But I can try something.